libfreerdp/core/transport.c

```diff
@@ -773,6 +838,82 @@ SSIZE_T transport_parse_pdu(rdpTransport* transport, wStream* s, BOOL* incomplet
 			}
 		}
 	}
+	else if (transport->RdstlsMode)
+	{
+		UINT16 version = (header[1] << 8) | header[0];
+		UINT16 pduType;
+		UINT16 dataType;
+
+		if (version != RDSTLS_VERSION_1)
+		{
+			WLog_Print(transport->log, WLOG_ERROR, "invalid RDSTLS version");
+			return -1;
+		}
+
+		if (position < 4)
+			return 0;
+
+		pduType = (header[3] << 8) | header[2];
+		switch (pduType)
+		{
+			case RDSTLS_TYPE_CAPABILITIES:
+				pduLength = 8;
+				break;
+			case RDSTLS_TYPE_AUTHREQ:
+				if (position < 6)
+					return 0;
+
+				dataType = (header[5] << 8 | header[4]);
```

Comment:
Definitely use a static stream and `Stream_Read_XXX` functions to do all that stuff.